File: tests/visualization/wavePacket3D.py
import subprocess
import numpy as np
import matplotlib.pyplot as plt
import tools
import style

# Get host and user name.
host_name = subprocess.getoutput("hostname")
user_name = subprocess.getoutput("whoami")

if "levante" in host_name:
  # Levante cluster
  data_path = "/scratch/b/" + user_name + "/PF/runs"
  reference_path = "/scratch/b/" + user_name + "/PF/pinc/reference"

elif "login" in host_name:
  # Goethe cluster
  data_path = "/scratch/atmodynamics/" + user_name + "/PF/runs"
  reference_path = "/scratch/atmodynamics/" + user_name + "/PF/pinc/reference"

else:
  # Local machine
  data_path = "../"
  reference_path = "../"

print("data_path =", data_path)
print("reference_path =", reference_path)

# Import data.
data = tools.ModelOutput(data_path + 'wavePacket3D/')
reference = tools.ModelOutput(reference_path + 'wavePacket3D/')

# Set slices.
it = - 1

print(" ".join(("Time:", str(data.variables['t'][it]), "s")))

# Set grid.
xx = data.variables['x'][:] * 0.001
zz = data.variables['z'][:] * 0.001

# Get density fluctuations.
rhop = data.groups['atmvar'].variables['rhop'][it, :, 0]
rhopref = reference.groups['atmvar'].variables['rhop'][it, :, 0]
deltarho = rhop - rhopref

# Create plot.
(levels, colormap) = style.symmetric_contours(rhop.min(), rhop.max())
(fig, axes) = plt.subplots()
plot = axes.contourf(xx, zz, rhop, levels, cmap = colormap)
axes.set_xlim(2000.0, 7000.0)
axes.set_ylim(10.0, 50.0)
axes.set_xlabel(r"$x \, \mathrm{\left[km\right]}$")
axes.set_ylabel(r"$z \, \mathrm{\left[km\right]}$")
fig.colorbar(plot, label = r"$\rho' \, \mathrm{\left[kg \, m^{- 3}\right]}$")
fig.savefig("".join((data_path, "/results/wavePacket3D.pdf")))
fig.savefig("".join((data_path, "/results/wavePacket3D.png")), dpi = 500)

# Create difference plot.
if data_path != reference_path:
  (levels, colormap) = style.symmetric_contours(deltarho.min(), deltarho.max())
  (fig, axes) = plt.subplots()
  plot = axes.contourf(xx, zz, deltarho, levels, cmap = colormap)
  axes.set_xlim(2000.0, 7000.0)
  axes.set_ylim(10.0, 50.0)
  axes.set_xlabel(r"$x \, \mathrm{\left[km\right]}$")
  axes.set_ylabel(r"$z \, \mathrm{\left[km\right]}$")
  fig.colorbar(plot, label = r"$\rho' \, \mathrm{\left[kg \, m^{- 3}\right]}$")
  fig.savefig("".join((data_path, "/results/wavePacket3D_difference.pdf")))
  fig.savefig("".join((data_path, "/results/wavePacket3D_difference.png")), \
      dpi = 500)

File: tests/visualization/wavePacket3D.jl
include("style.jl")
include("tools.jl")

using LaTeXStrings

# Set paths.
data_path = "../"
reference_path = "../"

println("data_path =", data_path)
println("reference_path =", reference_path)

# Import data.
data = ModelOutput(data_path * "wavePacket3D")
reference = ModelOutput(reference_path * "wavePacket3D")

println("Time: ", data.variables["t"][end], " s")

# Set grid.
x = data.variables["x"][120:390] .* 0.001
z = data.variables["z"][120:420] .* 0.001
x = [xi for xi in x, j in 1:size(z)[1]]
z = [zj for i in 1:size(x)[1], zj in z]

# Get density fluctuations.
rhop = data.groups["atmvar"].variables["rhop"][120:390, 1, 120:420, end]
rhopr = reference.groups["atmvar"].variables["rhop"][120:390, 1, 120:420, end]
deltarhop = rhop .- rhopr

# Create plot.
(levels, colormap) = symmetric_contours(minimum(rhop), maximum(rhop))
contours = contourf(x, z, rhop; levels = levels, cmap = colormap)
xlabel(L"x\,\left[\mathrm{km}\right]")
ylabel(L"z\,\left[\mathrm{km}\right]")
colorbar(contours; label = L"\rho'\,\left[\mathrm{kg\,m^{- 3}}\right]")
savefig(data_path * "/results/wavePacket3D.png")

if data_path != reference_path
  (levels, colormap) =
    symmetric_contours(minimum(deltarhop), maximum(deltarhop))
  clf()
  contours = contourf(x, z, deltarhop; levels = levels, cmap = colormap)
  xlabel(L"x\,\left[\mathrm{km}\right]")
  ylabel(L"z\,\left[\mathrm{km}\right]")
  colorbar(contours; label = L"\Delta\rho'\,\left[\mathrm{kg\,m^{- 3}}\right]")
  savefig(data_path * "/results/wavePacket3D_difference.png")
end
